src/zh/komiic/src/eu/kanade/tachiyomi/extension/zh/komiic/Komiic.kt

```diff
@@ -0,0 +1,291 @@
+                    else -> chapter.serial
+                }
+                date_upload = parseDate(chapter.dateCreated)
+                chapter_number = chapter.serial.toFloatOrNull() ?: -1f
+            }
+        }.reversed()
+
+        return entries
+    }
+
+    // Chapter list
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val payload = Payload(
+            operationName = "chapterByComicId",
+            variables = ChapterByComicIdVariables(manga.url.substringAfterLast("/")),
+            query = QUERY_CHAPTER,
+        ).toJsonRequestBody()
+
+        return client.newCall(POST(queryAPIUrl, headers, payload))
+            .asObservableSuccess()
+            .map { response ->
+                parseChapterList<ChaptersResponse>(response, manga.url)
+            }
+    }
+
+    override fun chapterListParse(response: Response) = throw UnsupportedOperationException()
+
+    /**
+     * 檢查 API 是否達到上限
+     * Check if the API has reached its limit.
+     *
+     * (Idk how to throw an exception in reading page)
+     */
+    private fun fetchAPILimit(): Boolean {
+        val payload = Payload("getImageLimit", "", QUERY_API_LIMIT).toJsonRequestBody()
+        val response = client.newCall(POST(queryAPIUrl, headers, payload)).execute()
+        val limit = response.parseAs<APILimitData>().getImageLimit
+        return limit.limit <= limit.usage
+    }
+
+    // Page list
+    override fun pageListParse(response: Response) = throw UnsupportedOperationException()
+
+    override fun fetchPageList(chapter: SChapter): Observable<List<Page>> {
+        val payload = Payload(
+            operationName = "imagesByChapterId",
+            variables = ImagesByChapterIdVariables(
+                chapter.url.substringAfter("/chapter/").substringBefore("/page/"),
+            ),
+            query = QUERY_PAGE_LIST,
+        ).toJsonRequestBody()
+        return client.newCall(POST(queryAPIUrl, headers, payload))
+            .asObservableSuccess()
+            .map { response ->
+                val res = response.parseAs<Data<ImagesResponse>>()
+                val pages = res.data.images
+
+                pages.mapIndexed { index, image ->
+                    Page(
+                        index,
+                        "${chapter.url.substringBeforeLast("/")}/page/$index",
+                        "https://komiic.com/api/image/${image.kid}",
+                    )
+                }
+            }
+    }
+
+    override fun imageRequest(page: Page): Request {
+        return super.imageRequest(page).newBuilder()
+            .addHeader("accept", "image/avif,image/webp,image/apng,image/svg+xml,image/*,*/*;q=0.8'")
+            .addHeader("referer", page.url)
+            .build()
+    }
+
+    override fun imageUrlParse(response: Response) = throw UnsupportedOperationException()
+
+    private inline fun <reified T> String.parseAs(): T =
+        Json.decodeFromString(this)
```

Comment:
```
private val json: Json by injektLazy()
```
put this somewhere in the class, (near the top) and use that instead of directly calling `Json`

Reply:
![image](https://github.com/user-attachments/assets/3e83537a-d2a5-4e8e-b82f-728b132885bd)